clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/provider/agent/DockerRegistryImageCachingAgent.groovy

```diff
@@ -158,23 +160,26 @@ class DockerRegistryImageCachingAgent implements CachingAgent, AccountAware, Age
             }
           }
         }
-        try {
-          creationDate = credentials.client.getCreationDate(repository, tag)
-        } catch (Exception e) {
-          log.warn("Unable to fetch tag creation date, reason: {} (tag: {}, repository: {})", e.message, tag, repository)
+
+        if (credentials.sortTagsByDate) {
+          try {
+            creationDate = credentials.client.getCreationDate(repository, tag)
+          } catch (Exception e) {
+            log.warn("Unable to fetch tag creation date, reason: {} (tag: {}, repository: {})", e.message, tag, repository)
+          }
         }
 
-        cachedTags[tagKey].with {
+        cachedTags << [(tagKey) : new DefaultCacheDataBuilder().tap {
```

Comment:
Both maps are `ConcurrentHashMap` initialized here: https://github.com/spinnaker/clouddriver/blob/7b8d93b354f9361c0c8b1bb0ece5254fc535a2a4/clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/cache/DefaultCacheDataBuilder.groovy#L34 . I'll update the code to be more java-like.

Reply:
Ok, I see that now.  Since we're depending on the maps being concurrent maps, I think the return type of `defaultCacheDataBuilderMap` should be change to a `ConcurrentMap`.  Otherwise, `DockerRegistryImageCachingAgent` is depending on an implementation detail of `DefaultCacheDataBuilder` rather than an explicit contract.